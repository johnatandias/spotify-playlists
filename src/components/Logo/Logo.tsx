import React from 'react';

interface Logo {
  width: number,
  height: number,
  color?: string,
  opacity?: number,
  className?: string
}

export const Logo = ({ width, height, color = "#1AB16A", opacity = 1, className }: Logo) => (
  <svg viewBox="0 0 700 700" width={width} height={height} opacity={opacity} className={className}>
    <defs>
      <clipPath id="_clipPath_sxS70SyISXKPYOUo6m0B2aYomjjF4cMZ">
        <rect width="700" height="700"/>
      </clipPath>
    </defs>
    <g clipPath="url(#_clipPath_sxS70SyISXKPYOUo6m0B2aYomjjF4cMZ)">
      <path
        d=" M 350.127 22.873 C 158.006 22.873 2.254 169.332 2.254 349.994 C 2.254 530.676 158.006 677.127 350.127 677.127 C 542.256 677.127 698 530.676 698 349.994 C 698 169.344 542.264 22.889 350.123 22.889 L 350.127 22.873 Z  M 509.659 494.675 C 503.411 504.312 490.06 507.332 479.846 501.461 C 398.156 454.511 295.335 443.905 174.244 469.917 C 162.575 472.433 150.944 465.554 148.285 454.585 C 145.614 443.612 152.896 432.675 164.594 430.175 C 297.109 401.686 410.776 413.948 502.469 466.636 C 512.687 472.535 515.928 485.074 509.655 494.683 L 509.659 494.675 Z  M 552.238 405.616 C 544.37 417.62 527.679 421.386 514.931 414.014 C 421.444 359.975 278.889 344.322 168.287 375.893 C 153.943 379.967 138.793 372.365 134.44 358.9 C 130.12 345.412 138.208 331.193 152.527 327.092 C 278.864 291.044 435.921 308.505 543.303 370.557 C 556.056 377.94 560.085 393.643 552.234 405.62 L 552.238 405.616 Z  M 555.894 312.865 C 443.759 250.243 258.796 244.485 151.75 275.036 C 134.56 279.939 116.382 270.814 111.173 254.649 C 105.964 238.477 115.659 221.395 132.861 216.481 C 255.742 181.402 460.018 188.18 589.101 260.239 C 604.563 268.868 609.635 287.646 600.471 302.165 C 591.332 316.705 571.31 321.498 555.898 312.865 L 555.894 312.865 Z "
        fill={color}
      />
    </g>
  </svg>
);
