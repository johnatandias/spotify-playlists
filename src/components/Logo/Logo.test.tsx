import React from "react";
import { shallow } from 'enzyme';
import { Logo } from './Logo';

describe('Logo component', () => {
  it('Renders without crashing', () => {
    const size = { width: 300, height: 300 };
    const wrapper = shallow(<Logo {...size} />);
    expect(wrapper.find(`[height=${size.height}]`)).toHaveLength(1);
    expect(wrapper.find(`[width=${size.width}]`)).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('Change logo opacity', () => {
    const props = { width: 300, height: 300, opacity: 0.3 };
    const wrapper = shallow(<Logo {...props} />);
    expect(wrapper.find(`[opacity=${props.opacity}]`)).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('Renders with classname', () => {
    const props = { width: 300, height: 300, className: 'my-class' };
    const wrapper = shallow(<Logo {...props} />);
    expect(wrapper.hasClass(props.className)).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
