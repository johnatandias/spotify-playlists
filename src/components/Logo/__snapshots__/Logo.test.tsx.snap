// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component Change logo opacity 1`] = `ShallowWrapper {}`;

exports[`Logo component Renders with classname 1`] = `ShallowWrapper {}`;

exports[`Logo component Renders without crashing 1`] = `ShallowWrapper {}`;
